force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -124,18 +138,23 @@ public with sharing class ClassesObjectsMethods {
     public static String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
-
+         // The variable to store the concatenated string
+        List<String> filteredList = new List<String>();
+        for(String scientist : scientistNames) {// Loop through the list of scientists' names
+        if(!scientist.containsIgnoreCase('g')){
+            continue;
             // If the name doesn't contain the letter 'G', skip this iteration
+        }
+        else {
+            filteredList.add(scientist);
+             // Add the name to the result string, followed by a comma
+        }
```

Comment:
YES finally someone uses this approach. String.join is your best friend and you should remember that method.